### PR TITLE
kubedock: init at 0.14.1

### PR DESCRIPTION
**Pull request by mausch**
## Description of changes

https://github.com/joyrex2001/kubedock
 Kubedock is a minimal implementation of the docker api that will orchestrate containers on a Kubernetes cluster, rather than running containers locally. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/kubedock/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    rev = gitref;
+    hash = "sha256-xjt/RqvkOHTSTZwNgxmlUeYndPW9jwUE1knLyNXDxaE=";
+  };
+
+  vendorHash = "sha256-4X4v40c4J5SPwj53Zk+j+A9k0RkHGwqz128E+FnLr1c=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/joyrex2001/kubedock/internal/config.Build=${gitref}"
```
Comment (pbsds) [inline pkgs/development/tools/kubedock/default.nix:22]: Upstream seems to have version tags, is the gitref strictly neccesary?

File: pkgs/development/tools/kubedock/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "-X github.com/joyrex2001/kubedock/internal/config.Version=${version}"
+  ];
+
+  CGO_ENABLED = 0;
+
+  meta = with lib; {
+    description = "Minimal implementation of the Docker API that will orchestrate containers on a Kubernetes cluster";
+    homepage = "https://github.com/joyrex2001/kubedock";
+    license = licenses.mit;
+    maintainers = with maintainers; [ mausch ];
```
Comment (pbsds) [inline pkgs/development/tools/kubedock/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ mausch ];
    mainProgram = "kubedock";
```

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 268352` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubedock</li>
  </ul>
</details>

Runs fine, some nits

File: pkgs/development/tools/kubedock/default.nix (hunk 1 above)
Comment (mausch) [reply pkgs/development/tools/kubedock/default.nix:22]: Not sure what you mean? this is what I get from running it on docker:
```
$ docker run --rm joyrex2001/kubedock version
-------------------------------------------------
kubedock
-------------------------------------------------
version: 0.14.1-7-g06b1305
date:    20231118-124310
build:   06b1305d9ffc318eca4e8f9f7840b38f71a6692e
-------------------------------------------------
```

That build value points to commit https://github.com/joyrex2001/kubedock/commit/06b1305d9ffc318eca4e8f9f7840b38f71a6692e

The build upstream sets this in https://github.com/joyrex2001/kubedock/blob/5aad204a72fcd45b1540ed443d7cfff2fed9ece0/Makefile#L2

Without this it prints `<undef>`

Comment (pbsds) [reply pkgs/development/tools/kubedock/default.nix:22]: Would `config.Build=refs/tags/${version}` work? The git ref makes it so r-ryantm can't do its job on this package

Comment (mausch) [reply pkgs/development/tools/kubedock/default.nix:22]: Ah I see. Unfortunately that doesn't work...
I see a few options here, none of them ideal:
1. Leave it as it is and forgo r-ryantm
2. Use `leaveDotGit` ([people say it has issues](https://discourse.nixos.org/t/keep-git-folder-in-when-fetching-a-git-repo/8590) though)
3. Leave `config.Build` undefined

Any other ideas? I'd go for option 3 here.

Comment (pbsds) [reply pkgs/development/tools/kubedock/default.nix:22]: Option 1 is also fine, if you add adequate reasoning in comments.

Comment (mausch) [reply pkgs/development/tools/kubedock/default.nix]: I went with option 3 in https://github.com/NixOS/nixpkgs/pull/268352/commits/6e51d84a14afc9597646a92be79bbe8ba7607c45

**Pull request merged**